sw/device/silicon_creator/lib/drivers/flash_ctrl.h

```diff
@@ -57,60 +57,97 @@ typedef enum flash_crtl_partition {
   ((bitfield_field32_t){.mask = 0x3, .index = 1})
 
 /**
- * Helper macro for defining the value of a `flash_ctrl_info_page_t` enumeration
- * constant for information pages of type 0.
+ * Table of flash information pages.
  *
- * Each `flash_ctrl_info_page_t` enumeration constant is a bitfield with the
- * following layout:
- * - Bits 0-3: Page index ([0,9] for type 0, 0 for type 1, [0,1] for type 2).
- * - Bits 4-6: Partition type (a `flash_ctrl_partition_type_t`).
- * - Bit 7: Bank index [0,1].
+ * Columns: Name, value, bank index, page index.
+ * We use an X macro to faciliate writing enums, swtich statements, and unit
+ * tests using the contants here. All information pages in this table are of
+ * type 0 since silicon creator code does not need to access information pages
+ * of other types.
+ *
+ * Encoding generated with
+ * $ ./util/design/sparse-fsm-encode.py -d 6 -m 20 -n 32 \
+ *     -s 1755363476 --language=c
  *
- * This macro assumes that all information pages are of type 0 since silicon
- * creator code does not need to access information pages of other types.
+ * Hamming distance histogram:
  *
- * @param bank_ Bank index.
- * @param page_ Page index.
+ *  0: --
+ *  1: --
+ *  2: --
+ *  3: --
+ *  4: --
+ *  5: --
+ *  6: --
+ *  7: --
+ *  8: --
+ *  9:  (0.53%)
+ * 10: || (2.63%)
+ * 11: || (2.11%)
+ * 12: |||||| (6.84%)
+ * 13: |||||| (6.84%)
+ * 14: ||||||||||| (12.11%)
+ * 15: ||||||||||| (11.58%)
+ * 16: |||||||||||||||||||| (20.53%)
+ * 17: ||||||||| (10.00%)
+ * 18: |||||||||| (10.53%)
+ * 19: ||||||| (7.37%)
+ * 20: |||||| (6.84%)
+ * 21: | (1.05%)
+ * 22: | (1.05%)
+ * 23: --
+ * 24: --
+ * 25: --
+ * 26: --
+ * 27: --
+ * 28: --
+ * 29: --
+ * 30: --
+ * 31: --
+ * 32: --
+ *
+ * Minimum Hamming distance: 9
+ * Maximum Hamming distance: 22
+ * Minimum Hamming weight: 13
+ * Maximum Hamming weight: 25
  */
-#define INFO_PAGE_(bank_, page_) \
-  ((bank_ << 7) | (kFlashCtrlPartitionInfo0 << 4) | (page_))
-
 // clang-format off
 #define FLASH_CTRL_INFO_PAGES_DEFINE(X) \
   /**
    * Bank 0 information partition type 0 pages.
    */ \
-  X(kFlashCtrlInfoPageFactoryId,          0, 0), \
-  X(kFlashCtrlInfoPageCreatorSecret,      0, 1), \
-  X(kFlashCtrlInfoPageOwnerSecret,        0, 2), \
-  X(kFlashCtrlInfoPageWaferAuthSecret,    0, 3), \
-  X(kFlashCtrlInfoPageBank0Type0Page4,    0, 4), \
-  X(kFlashCtrlInfoPageBank0Type0Page5,    0, 5), \
-  X(kFlashCtrlInfoPageOwnerReserved0,     0, 6), \
-  X(kFlashCtrlInfoPageOwnerReserved1,     0, 7), \
-  X(kFlashCtrlInfoPageOwnerReserved2,     0, 8), \
-  X(kFlashCtrlInfoPageOwnerReserved3,     0, 9), \
+  X(kFlashCtrlInfoPageFactoryId,         	0x9dc41c33, 0, 0) \
```

Comment:
I asked the same question in a CL from jade, but do we want these to be 12-bit or 32-bit, for the purposes of fitting into an immediate?

Reply:
I think it's ok to use 32-bit values here since these are flash_ctrl function arguments which are used in a couple of places in the code. Also, the `sparse-fsm-encode` requires the width to be a power of two.